core/jazz_create-serverless-service/index.js

```diff
@@ -351,6 +351,19 @@ var getServiceData = (service_creation_data, authToken, configData, deploymentTa
     if (service_creation_data.service_type === "api" || service_creation_data.service_type === "function" || service_creation_data.service_type === "sls-app") {
       serviceMetadataObj.providerRuntime = service_creation_data.runtime;
     }
+    
+    //Adding providerTimeout key in service catalog
+    if (service_creation_data.service_type === "api" || service_creation_data.service_type === "function") {
+      if (service_creation_data.timeout)
+      {
+        serviceMetadataObj.providerTimeout = service_creation_data.timeout;
+      }
+      else
+      {
+        // default to 160 seconds
+        serviceMetadataObj.providerTimeout = 160;
```

Comment:
Move this to configuration. Also, 30 sec is a good start (API Gateway max time out is 29 sec).

Reply:
Updated.